jax/custom_derivatives.py

```diff
@@ -208,27 +183,39 @@ def __call__(self, *args, **kwargs):
       raise AttributeError(msg.format(self.__name__))
     args = _resolve_kwargs(self.fun, args, kwargs)
     if self.nondiff_argnums:
-      is_nondiff = [False] * len(args)
-      for i in self.nondiff_argnums: is_nondiff[i] = True
-      args = [_stop_gradient(x) if b else x for b, x in zip(is_nondiff, args)]
-      dyn_argnums = [i for i, b in enumerate(is_nondiff) if not b]
-      f_, dyn_args = argnums_partial(lu.wrap_init(self.fun), dyn_argnums, args)
+      args = [_stop_gradient(x) if i in self.nondiff_argnums else x
```

Comment:
Isn't that quadratic in the number of arguments? The previous implementation is linear. Maybe make `nondiff_argnums` a set before this line?

Reply:
It is but I thought it wouldn't matter :) I guess some functions have lots of arguments! I'll fix this.